src/main/java/com/somesoft/fittracker/service/ProductService.java

```diff
@@ -49,6 +54,15 @@ public ProductResponse update(UUID id, ProductRequest productRequest) {
             throw new ProductNotUpdatedException();
         }
 
+        Optional<Product> duplicateProduct = productRepository.findByNameAndUserIdAndActiveIsTrue(
+            updatedProduct.getName(), SecurityHelper.getUserId());
+
+        duplicateProduct.ifPresent(product -> {
+            if (!product.getId().equals(id)) {
+                throw new ProductAlreadyExistsException(updatedProduct.getName());
+            }
+        });
+
```

Comment:
Let's try to use optionals when possible for readability + I'd extract it to private method as above. My suggestion:
```
private boolean productAlreadyExists(UUID id, Product product) {
    return productRepository.findByNameAndUserIdAndActiveIsTrue(product.getName(), SecurityHelper.getUserId())
        .map(Product::getId)
        .filter(foundProductId -> foundProductId.equals(id))
        .isEmpty();
}
```
that way we could replace this whole block with similar code as for save (but with id):
```
if (productAlreadyExists(id, product)) {
    throw new ProductAlreadyExistsException(updatedProduct.getName());
}
```

Also, let's rename `updatedProduct` to just `product`, we know it is new, updated version in the context of `update` method. And adding `updated` to the name might suggest that the original user request has been manipulated in some way.